ga4gh/datamodel/__init__.py

```diff
@@ -135,22 +136,58 @@ def __init__(self, parentCompoundId, *localIds):
                 setattr(self, field, getattr(parentCompoundId, field))
                 index += 1
         for field, localId in zip(self.fields[index:], localIds):
-            setattr(self, field, str(localId))
+            encodedLocalId = self.encode(str(localId))
+            setattr(self, field, encodedLocalId)
         if len(localIds) != len(self.fields) - index:
             raise ValueError(
                 "Incorrect number of fields provided to instantiate ID")
         for idFieldName, prefix in self.containerIds:
             values = [getattr(self, f) for f in self.fields[:prefix + 1]]
-            containerId = self.separator.join(values)
+            containerId = self.join(values)
             obfuscated = self.obfuscate(containerId)
             setattr(self, idFieldName, obfuscated)
 
     def __str__(self):
         values = [getattr(self, f) for f in self.fields]
-        compoundIdStr = self.separator.join(values)
+        compoundIdStr = self.join(values)
         return self.obfuscate(compoundIdStr)
 
     @classmethod
+    def join(cls, splits):
+        """
+        Join an array of ids into a compound id string
+        """
+        segments = []
+        for split in splits:
+            segments.append('"{}", '.format(split))
```

Comment:
I think drop the trailing space here, it's just inflating the size of IDs, which already have a lot of padding.